packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -3516,6 +3516,9 @@ class NavigatorState extends State<Navigator> with TickerProviderStateMixin, Res
 
   bool get _usingPagesAPI => widget.pages != const <Page<dynamic>>[];
 
+  /// Whether the route was started popping by user gesture.
+  bool popGestureStarted = false;
```

Comment:
This is a confusing name, should probably called inAutomatedPopGesture, or finalizingPopGesture

Reply:
This property contradict the `userGestureInProgressNotifier`, which is unfortunate.

It looks like `userGestureInProgressNotifier` needs a new name like `userGestureDrivenPopInProgress`, or update its logic to match its name. The former will be a breaking change, the latter will require making cupertinoPageTransition more resilient (See https://github.com/flutter/flutter/blob/5a41e1edac9f4055f78bd38b3833bbc1f5a2b294/packages/flutter/lib/src/widgets/routes.dart#L543)